### ISSUE TITLE
Scope `removeContext` and `removeTag` not working for app crash

### ISSUE THREAD

**Issue by denrase** [getsentry/sentry-cocoa#2588]
### Platform

iOS

### Installed

Swift Package Manager

### Version

8.0.0-beta4

### Steps to Reproduce

1. Set a context key with data and remove it again.

```
SentrySDK.configureScope { scope in
    scope.setContext(value: ["key": "value"], key: "context-key")
    scope.setContext(value: ["key-2": "value-2"], key: "context-key-2")
    scope.removeContext(key: "context-key-2")
}
```

2. Launch application and trigger a hard crash. This was done with the swift sample app.
3. Launch application again in order for the crash to sync to the backend.

Same behaviour for set/remove tag.

### Expected Result

Only data  for `context-key` is present.

### Actual Result

Data  for `context-key-2` is still present.

<img width="366" alt="Bildschirmfoto 2023-01-09 um 14 01 36" src="https://user-images.githubusercontent.com/3984453/211315861-6f05152d-5ba0-4fcf-a8d8-dcd19d3f8123.png">

**Comment by philipphofmann** [comment]
The problem is here 
https://github.com/getsentry/sentry-cocoa/blob/e6fd1fba83e54dc9cf037cc9325736e1b4f2f48e/Sources/Sentry/SentryScope.m#L226-L235

It should be `setContext` instead of `setExtras`.

**Comment by marandaneto** [comment]
@philipphofmann can we fix that and cut a patch for v7? Flutter v7 with Cocoa v8 needs more time.

**Comment by philipphofmann** [comment]
> @philipphofmann can we fix that and cut a patch for v7? Flutter v7 with Cocoa v8 needs more time.

Is it that important? Of course, we can, it's just extra work.

**Comment by philipphofmann** [comment]
> @philipphofmann can we fix that and cut a patch for v7? Flutter v7 with Cocoa v8 needs more time.

Is it that important? Of course, we can, it's just extra work.

**Comment by marandaneto** [comment]
It's a bug that affects the Hybrid SDKs + the iOS SDK, it's not a p0 but definitely, a bug, imagine that you are trying to remove some PII and you can't, then it's more severe.

**Comment by brustolin** [comment]
Since this looks like an edge case that we don't have any complains from customer, we gonna fix this only for v8. Unless someone has a strong opinion against it.

**Comment by philipphofmann** [comment]
> imagine that you are trying to remove some PII and you can't, then it's more severe.

If you once add any PII to the context or the tags and you remove it directly afterward, it still could end up in Sentry. Therefore, the PII argument is weaker for me. You should never add any PII to the context or tags.

**Comment by marandaneto** [comment]
> > imagine that you are trying to remove some PII and you can't, then it's more severe.
> 
> If you once add any PII to the context or the tags and you remove it directly afterward, it still could end up in Sentry. Therefore, the PII argument is weaker for me. You should never add any PII to the context or tags.

I agree with the statement that you should not add PII, but IMO that's another problem, one can add it already, so one should be able to remove it, I'm focused on the side effect of the bug and not the intention of the user.

**Comment by philipphofmann** [comment]
> I agree with the statement that you should not add PII, but IMO that's another problem, one can add it already, so one should be able to remove it, I'm focused on the side effect of the bug and not the intention of the user.

That's a good point, but we don't think we must do a hotfix. We definitely are going to fix this, but we think it's sufficient to do this in 8.x.x.

**Comment by philprime** [comment]
This issue has not had any activity in 2 years. Is this still an issue in the latest v8.x.x releases?

**Comment by philipphofmann** [comment]
The problem still exists. The `removeContextForKey` didn't change in the Scope. This should be an easy fix.
